cmd/clusterctl/clusterdeployer/clusterdeployer_test.go

```diff
@@ -284,6 +284,15 @@ func (c *testClusterClient) DeleteClusters(ns string) error {
 	} else {
 		delete(c.clusters, ns)
 	}
+	if err := c.DeleteMachineDeployments(ns); err != nil {
```

Comment:
(Note, this isn't a comment about your changes, but one about the way this test was originally written 😄.) We should consider adjusting the test and its verification logic, because `TestClusterDelete` is really testing behavior of the `testClusterClient`. It seems off that your fix—remove the calls to delete MachineDeployments, MachineSets, and Machines—requires that you **add** these deletions to the `testClusterClient`. cc @chuckha

Reply:
Or maybe a change you could make in this PR is to remove any expectation in the test code around MachineSets, MachineDeployments, and Machines. You could remove the checks for their deletions, and you could remove the test cases that involve errors creating and deleting these 3 resources.

Reply:
That sounds reasonable to me.

Reply:
I've removed the tests and validations that don't make sense anymore.